tuf/download.py

```diff
@@ -278,7 +278,7 @@ def _download_file(url, required_length, STRICT_REQUIRED_LENGTH=True):
   except Exception:
     # Close 'temp_file'.  Any written data is lost.
     temp_file.close()
-    logger.exception('Could not download URL: ' + repr(url))
+    logger.debug('Could not download URL: ' + repr(url))
```

Comment:
Can you elaborate on why you chose debug level here?

Reply:
I don't have a strong opinion here, but I think the reason I went with debug() was to avoid info() for things that 
* do happen in normal usage
* are then raised as exceptions and handled in calling code: in this case Updater will usually log 'Update failed from ...' also at debug level

Reply:
just so I'm not misleading: 'Update failed from ...' log level was also chosen by my in this PR.

The point that I tried to make was that the 'lower level code' (download) probably should not log at INFO if the 'higher level code' (Updater) logs the same thing at DEBUG. A case could be made for the Updater logs for failed updates to be INFO though...

Reply:
And that logging in updater is only a DEBUG because an exception is raised if all mirrors fail. This makes sense, thanks for providing that extra context.

I agree, a case could be made for info. But I'm not making that case here.